plugins/modules/filesystem.py

```diff
@@ -197,6 +198,125 @@ def fs_state(module, filesystem):
     return False
 
 
+def check_attr_change(module, filesystem):
+    """
+    Determines if changes will be made on the filesystem.
+    param module: Ansible module argument spec.
+    param filesystem: filesystem name.
+    return: True - changes will be made on the filesystem / False = filesystem will remain unchanged
```

Comment:
Can we add more details in document to state which attributes will be checked, which attributes will be passed

Reply:
`  device:
    description:
    - When I(state=present), for local filesystem, it specifies the logical volume to use to create
      the filesystem. If not specified, a new logical volume will be created.
    - When I(state=present), for NFS filesystem, it specifies the remote export device to use to
      create or modify the filesystem.
    type: str
`
`  nfs_server:
    description:
    - Specifies a Network File System (NFS) server for NFS filesystem.
    type: str
`

Reply:
they do say how those parameters are treated for NFS. but i can add additional text to say that if you are attempting to modify an NFS filesystem then device and nfs_server are required.